### PR TITLE
Organizational and cleanup tweaks

### PR DESCRIPTION
**Pull request by nwiltsie**
# Description

This PR performs a handful of re-organizational and cleanup tasks for the pipeline.

### Sidecar file searching during config

I've added the following "new" required parameters to `default.config` and `schema.yaml`. They all have default values defined by other parameters, e.g. `reference_fasta_fai = "${-> params.reference_fasta}.fai"`.

* `reference_fasta_fai`
* `reference_fasta_dict`
* `bundle_known_indels_vcf_gz_tbi`
* `bundle_contest_hapmap_3p3_vcf_gz_tbi`
* `bundle_mills_and_1000g_gold_standard_indels_vcf_gz_tbi`
* `bundle_v0_dbsnp138_vcf_gz_tbi`

I say "new" parameters because the values are already in use across the pipeline - this change is to eagerly perform the discovery and validation of those files during the configuration phase. That's more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and ensures that missing files cause fast failures.

### Optional parameters listed in README

I've updated the README to include all of the parameters from `default.config` in a second table.

### PipeVal in the critical path

I've tweaked the early workflow processes to take the PipeVal validated output files rather than the raw BAM/BAI files (required https://github.com/uclahs-cds/pipeline-Nextflow-module/pull/44). That trades a little efficiency for simplicity - the downstream processing can't begin in parallel with validation, but in exchange we don't need to worry about cancelling processes or clawing back invalid outputs due to a late validation result.

### Input name harmonization

Many processes have proxy inputs for parameters - that is, those processes are always called with the same parameter as the same positional input. In those cases I renamed the input to match the parameter - for example, `run_SplitIntervals_GATK`'s `reference`, `reference_index`, and `reference_dict` inputs are now `reference_fasta`, `reference_fasta_index`, and `reference_fasta_dict`.

Harmonizing the input name with the parameter name makes it easier to trace the logic and search the codebase.

## Testing Results

- NFTest
  - log:      `/hot/software/pipeline/pipeline-recalibrate-BAM/Nextflow/development/unreleased/nwiltsie-refactor/log-nftest-20240627T155900Z.log`
  - cases:    default set

# Checklist

- [x] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have reviewed the [Nextflow pipeline standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3193890/Nextflow+pipeline+standardization).

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using \[AD\_username (or 5 letters of AD if AD is too long)]-\[brief\_description\_of\_branch\].

- [x] I have set up or verified the branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [x] I have added my name to the contributors listings in the ``manifest`` block in the `nextflow.config` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

- [x] I have added the changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

- [x] I have updated the version number in the `metadata.yaml` and `manifest` block of the `nextflow.config` file following [semver](https://semver.org/), or the version number has already been updated. (*Leave it unchecked if you are unsure about new version number and discuss it with the infrastructure team in this PR.*)

- [x] I have tested the pipeline using NFTest, _or_ I have justified why I did not need to run NFTest above.

### CODE REVIEW

**Review by tyamaguchi-ucla — APPROVED** [review]
Amazing work! Looks good to me. @yashpatel6 for the final approval.

File: config/default.config [hunk 1]
```diff
@@ -30,6 +32,14 @@ params {
     docker_image_samtools = "${-> params.docker_container_registry}/samtools:${params.samtools_version}"
 
     gatk_ir_compression = 1
+
+    // These parameters are inferred from the input files. The user can override them in the config file if required.
+    reference_fasta_fai = "${-> params.reference_fasta}.fai"
+    reference_fasta_dict = "${-> Nextflow.file(params.reference_fasta).resolveSibling(Nextflow.file(params.reference_fasta).getBaseName() + '.dict')}"
```
Comment (yashpatel6) [inline config/default.config:38]: **question (non-blocking)**: Any reason to use `resolveSibling` for the dict but not for the other files?

**Review by yashpatel6 — APPROVED** [review]
Looks good!

File: config/default.config (hunk 1 above)
Comment (nwiltsie) [reply config/default.config:38]: The other files append to the file extension (e.g. `reference.fasta.fai`) but the dictionary replaces it (e.g. `reference.dict`).

https://github.com/uclahs-cds/pipeline-recalibrate-BAM/blob/25a0b8e534de2addff748c744b84b34375248ab5/test/configtest-F16.json#L231-L233

Comment (yashpatel6) [reply config/default.config:38]: Ah right, it's to keep the full path intact since basename only gives the filename

**Pull request merged**